src/app/editor.js

```diff
@@ -100,9 +110,15 @@ function Editor (loadingFromGist) {
   };
 
   this.clearAnnotations = function () {
+    sourceAnnotations = [];
     editor.getSession().clearAnnotations();
   };
 
+  this.addAnnotation = function (annotation) {
+    sourceAnnotations[sourceAnnotations.length] = annotation;
+    this.setAnnotations(sourceAnnotations);
+  };
+
   this.setAnnotations = function (sourceAnnotations) {
     editor.getSession().setAnnotations(sourceAnnotations);
```

Comment:
Sorry, can you clarify? Which array should be updated? Are you suggesting extracting annotations behind its own interface?